packages/dds/tree/src/feature-libraries/chunked-forest/chunkTree.ts

```diff
@@ -159,19 +160,19 @@ export class Chunker implements IChunker {
  *
  * @param cursor - cursor in nodes mode
  */
-export function chunkTree(cursor: ITreeCursorSynchronous, policy: ChunkPolicy): TreeChunk {
-	return chunkRange(cursor, policy, 1, true)[0] ?? oob();
+export function chunkTree(cursor: ITreeCursorSynchronous, policy: ChunkPolicy, idCompressor?: IIdCompressor): TreeChunk {
```

Comment:
Having to pass multiple things through a bunch of functions like this can be kindof annoying, and when one (or more) of them are option, its really fragile, since if any of the code paths forget to pass the optional one down, the code still builds and mostly works.

A design pattern for cases like this that is more robust is to group all the things you need to pass to all these places into a single object, and pass that around.

Before you change, the ChunkPolicy was filling this roll.

THis suggests two approaches that might work well here:
1. Make a new object that holds both the chunk policy and the new data (either as two separate fields of an aggregator object, or as a single object with both sets of data like `{...chunkPolicy, idCompressor}`
2. extend ChunkPolicy so that it has all the needed data.

I think either of these approaches would be fine here, but maybe its useful to be able to have a distinct type for a policy that does not have a compressor (ex: for things like defaultChunkPolicy), some maybe the fiirst option, and creating a new type, maybe we call it  ChunkCompressor thats `{policy: ChumPolicy, idCompressor: IIdCompressor | undefined }`

Reply:
Created new `ChunkCompressor` type, and used the default `ChunkPolicy` type for places where it was unnecessary (functions that don't handle `UniformChunks`)